### ISSUE TITLE
Accessible annotation icons and their tooltip

### ISSUE THREAD

**Issue by markov00** [elastic/elastic-charts#1942]
We are using a `button` element to enclose an annotation icon.
This button is visible from the screen readers but doesn't show a tooltip and doesn't have any description.
We should improve it by showing a readable tooltip when you focus on the icon or don't use a `button` type.

**Kibana Cross Issues**
https://github.com/elastic/kibana/issues/148684
